src/lib/dlt_client.c

```diff
@@ -1246,9 +1251,12 @@ DltReturnValue dlt_client_parse_get_log_info_resp_text(DltServiceGetLogInfoRespo
             /* get con id */
             dlt_getloginfo_conv_ascii_to_id(rp,
                                             &rp_count,
-                                            con->context_id,
+                                            temp_id,
```

Comment:
This change should not be required as dlt_getlonginfo_conv_ascii_to_id() takes care of the argument DLT_ID_SIZE. Please revert to original code.

Reply:
Hello! As described in the added code comment, `dlt_getlonginfo_conv_ascii_to_id()` does indeed have a length parameter, but the issue is that it also writes an **additional** NULL terminator after writing the specified `len` bytes. Since the `con->context_id` and `app->app_id` char arrays are of size **4**, the method will fill the buffer and will write another byte after the end of the ID array, **overwriting** one byte of the variable that resides next to those in their respective structures.

In other usages (App/Ctx descriptions), an additional byte is allocated specifically for the NULL terminator.